bot/decorators.py

```diff
@@ -9,49 +9,41 @@
 from discord.ext.commands import Context
 
 from bot.constants import ERROR_REPLIES
+from bot.utils.checks import in_channel_check, with_role_check, without_role_check
 
 log = logging.getLogger(__name__)
 
 
 def with_role(*role_ids: int):
```

Comment:
I've been wondering - why do we have our own decorator for this? Why not https://discordpy.readthedocs.io/en/rewrite/ext/commands/api.html#discord.ext.commands.has_any_role?

Reply:
@jchristgit that's a good question, so I had a look.

Turns out, the reason we wrote this decorator was because up until about a month ago, `has_any_role` did not support IDs, only role names.

It was added in this commit: https://github.com/Rapptz/discord.py/commit/016963500b6f19b5c8978f135fce9dd264872007

In other words, when we wrote it, it was superior to the thing you linked. But now, it's more or less identical.